c++/src/Reader.cc

```diff
@@ -41,6 +41,8 @@ namespace orc {
     "1.6.0", "1.6.1", "1.6.2", "1.6.3", "1.6.4", "1.6.5", "1.6.6", "1.6.7", "1.6.8",
     "1.6.9", "1.6.10", "1.6.11", "1.7.0"};
 
+  static const RowReaderOptions::IdReadIntentMap EMPTY_ID_READ_INTENT_MAP;
```

Comment:
Actually, when I test this PR on my Mac. `Clang 13` complains with `declaration requires an exit-time destructor` and this is a regression although this is a warning. Could you try `Clang 13` and remove the warning?
```
$ make package
[  5%] Built target snappy_ep
[ 11%] Built target zlib_ep
[ 17%] Built target zstd_ep
[ 23%] Built target lz4_ep
[ 29%] Built target googletest_ep
[ 35%] Built target protobuf_ep
Consolidate compiler generated dependencies of target orc
[ 35%] Building CXX object c++/src/CMakeFiles/orc.dir/Reader.cc.o
/Users/dongjoon/APACHE/orc-merge/c++/src/Reader.cc:44:50: error: declaration requires an exit-time destructor [-Werror,-Wexit-time-destructors]
  static const RowReaderOptions::IdReadIntentMap EMPTY_ID_READ_INTENT_MAP;
                                                 ^
/Users/dongjoon/APACHE/orc-merge/c++/src/Reader.cc:44:50: error: declaration requires a global destructor [-Werror,-Wglobal-constructors]
2 errors generated.
make[2]: *** [c++/src/CMakeFiles/orc.dir/Reader.cc.o] Error 1
make[1]: *** [c++/src/CMakeFiles/orc.dir/all] Error 2
make: *** [all] Error 2
```

Reply:
I replaced it with function `emptyIdReadIntentMap()`